packages/uninstaller/src/gcp.ts

```diff
@@ -36,31 +36,37 @@ export function* destroyGCPInfra(): Generator<
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   any
 > {
+  if (!destroyConfig.gcpProjectID) {
+    throw Error("gcp project id not specified in credentials");
```

Comment:
This error message isn't quite true, it would in fact be misleading when this would blow up. When this blows up it indicates a "programmer error", whereas the error message with "in credentials" suggests an operational/user error :)

I think this should be a simple assertion in code (based on NodeJS' assert module in strict mode (which tsc understands)) that will lead to a proper error message:

```
import { strict as assert } from "assert";
...
assert(destroyConfig.gcpProjectID);
...
<here, tsc knows, that `destroyConfig.gcpProjectID` is not undefined / falsy>
```

Reply:
Oh, and here is another example with an explicit comment:

https://github.com/opstrace/opstrace/blob/45f8cf57264fe3fb5d59a86744d14d38db9a60a2/packages/installer/src/aws.ts#L188

```
  // done assert instead of N non-null-assertion.
  assert(internetGateway.InternetGatewayId);
```